docs/internals/stream-corking.md

```diff
@@ -0,0 +1,254 @@
+# Stream Corking
+
+As discussed elsewhere, the Elastic Node.js Agent's HTTP Client is a writable stream object.  One advanced streaming feature the client uses [is Node.js stream corking](https://nodejs.org/api/stream.html#stream_writable_cork).
+
+We use stream corking to solve two independent challenges.  The first is to delay writes to the downstream stream chopper object until the client is 100% ready to send data.  The second is to prevent what Node.js calls "head-of-line blocking" issues, when the writing of individual objects downstream can creating blocking code situations.
```

Comment:
AFAIK this isn't "head of line blocking" https://en.wikipedia.org/wiki/Head-of-line_blocking

Reply:
_nod_ I was simply aping what the node docs themselves said. This will be easy enough to take out. https://nodejs.org/api/stream.html